evaluate.py

```diff
@@ -83,8 +83,10 @@ def main(
     ):
         print("Loading model ...", file=sys.stderr)
         t0 = time.time()
-        model = LLaMA.from_name(model_size)
         checkpoint = torch.load(checkpoint_path)
+        model = LLaMA.from_name(
+            llama_model_lookup[checkpoint["lm_head.weight"].shape[1]]
```

Comment:
I would turn `llama_model_lookup` into a function in `utils` that takes the whole checkpoint and returns the name. This way we can encapsulate the complete logic there, and just call

```
name = llama_model_lookup(checkpoint)
model = LLaMA.from_name(name)
```

Reply:
done